Modules/MUON/Common/src/TrackPlotter.cxx

```diff
@@ -0,0 +1,339 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "MUONCommon/TrackPlotter.h"
+
+#include "CommonConstants/LHCConstants.h"
+#include <DataFormatsMCH/Cluster.h>
+#include <DataFormatsMCH/Digit.h>
+#include <DataFormatsMCH/ROFRecord.h>
+#include <DataFormatsMCH/TrackMCH.h>
+#include <DetectorsBase/GeometryManager.h>
+#include <Framework/DataRefUtils.h>
+#include <Framework/InputRecord.h>
+#include <MCHMappingInterface/Segmentation.h>
+#include <MCHTracking/TrackExtrap.h>
+#include <MCHTracking/TrackParam.h>
+#include <Math/Vector4D.h>
+#include <Math/Vector4Dfwd.h>
+#include <ReconstructionDataFormats/TrackMCHMID.h>
+#include <TH1F.h>
+#include <TMath.h>
+#include <gsl/span>
+#include <set>
+
+namespace
+{
+
+constexpr double muonMass = 0.1056584;
+constexpr double muonMass2 = muonMass * muonMass;
+
+static void setXAxisLabels(TProfile* h)
+{
+  TAxis* axis = h->GetXaxis();
+  for (int i = 1; i <= 10; i++) {
+    auto label = fmt::format("CH{}", i);
+    axis->SetBinLabel(i, label.c_str());
+  }
+}
+
+uint16_t computeDsBinX(int feeId, int linkId, int elinkId)
+{
```

Comment:
It might be worth moving the function in the helpers, since we use the same kind of binning in other tasks.

Reply:
yes, but for another PR though...